src/Navbar.js

```diff
@@ -93,13 +124,19 @@ class Navbar extends React.Component {
   }
 
   getChildContext() {
-    const { bsClass, expanded } = this.props;
+    const { bsClass, expanded, onSelect, toggleOnSelect } = this.props;
+    const mobileToggle = () => {
+      if (typeof(window) !== 'undefined' && window.innerWidth < 768) {
```

Comment:
I agree - users shouldn't have to set a breakpoint.

I think I found a solution to this problem. It goes back to the point you mentioned about toggling vs closing the navbar! We were tackling it from the wrong angle with `toggleOnSelect`, which is causing the navbar to toggle between `expanded` states even when the screen size is not appropriate. In fact, screen size should not matter - only the value of `expanded` should really matter.

The right functionality should be something more like a `closeOnSelect` or `collapseOnSelect`, which will set `expanded` to false, only if it was true. If `expanded` is false, then it does nothing. I'll update this PR with a commit to demonstrate this new approach shortly.